### PR TITLE
Fix queue length excluding gpu kernels without parent event

### PR DESCRIPTION
**Pull request by briancoutinho**
## What does this PR do?
Fixes issue where queue lengths were negative

## Before submitting

- [ ] Was this discussed/approved via a Github issue? (no need for typos, doc improvements)
  - [x] N/A
- [x] Did you write any new necessary tests?
  - [ ] N/A
- [ ] Did you make sure to update the docs?
  - [x] N/A
- [ ] Did you update the [changelog](https://github.com/facebookresearch/HolisticTraceAnalysis/blob/main/CHANGELOG.md)?
  - [ ] N/A ??

## Tests
Checked on a trace where negative values show up- 
These are now fixed -

Screenshot 2023-07-17 at 8 13 49 PM
Screenshot 2023-07-18 at 10 28 02 AM

### CODE REVIEW

File: hta/analyzers/trace_counters.py [hunk 1]
```diff
@@ -65,20 +65,24 @@ def _get_queue_length_time_series_for_rank(
         gpu_kernels = trace_df[trace_df["stream"].ne(-1)].copy()
         gpu_kernels["queue"] = -1
 
+        # use the pid, tid and cuda stream from the correlated GPU event.
+        rutime_calls_filt = runtime_calls.join(
```
Comment (louisfeng) [inline hta/analyzers/trace_counters.py:69]: `rutime` -> `runtime`?

**Review by louisfeng — APPROVED** [review]
LGTM.

**Comment by anupambhatnagar** [comment]
@briancoutinho can we also add the trace I provided and add test cases for it. The traces we had for our existing test cases didn't cover that case.

**Comment by anupambhatnagar** [comment]
Here's the trace file that we can use to add test cases against negative queue length values.
[negative_queue_length_values.json.gz](https://github.com/facebookresearch/HolisticTraceAnalysis/files/12087910/negative_queue_length_values.json.gz)

**Review by anupambhatnagar — COMMENTED** [review]
Thanks for the quick fix. Let's add some more tests to safeguard against this case and then push it to main.
